### PR TITLE
improve code coverage for plugin/pkg/scheduler/schedulercache/node_info.go

### PR DESCRIPTION
**Pull request by du2016**
improve code coverage for plugin/pkg/scheduler/schedulercache/node_info.go for #39559

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
+package schedulercache
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"reflect"
+	"testing"
```
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Please invert the ordering of imports according to other files.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+var (
+	taints = []v1.Taint{
+		v1.Taint{
+			Key:   "foo",
+			Value: "bar",
+		},
+	}
+	testresource = Resource{1, 1, 1, map[v1.ResourceName]int64{"testresource": 1}}
+	affnity      = v1.Affinity{
```
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: s/affnity/affinity

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
+package schedulercache
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"reflect"
+	"testing"
+)
+
+var (
```
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Not a big fan of having a global across tests vs. an init f(n) that passes back an object that is used in the tests.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+	podwithoutaffinity := makepod("foo", "", &v1.Affinity{})
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithaffinity), true) {
+		t.Fatalf("nodeinfo check pod has podaffinityconstraints error,expected: %v,got: %v", true, hasPodAffinityConstraints(podwithaffinity))
+	}
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithoutaffinity), false) {
+		t.Fatalf("nodeinfo check pod has no podaffinityconstraints error,expected: %v,got: %v", false, hasPodAffinityConstraints(podwithoutaffinity))
+	}
+}
+
+func TestaddPod(t *testing.T) {
```
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: I would probably collapse the tests add + set + remove into a single test that verifies in that order, as it's a more realistic use case.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+	nodeinfo := makenodeinfo(node, taints)
+	pod := makepod("foo", "", &v1.Affinity{})
+	pods := []*v1.Pod{pod}
+	nodeinfo.pods = pods
+	if !reflect.DeepEqual(nodeinfo.Pods(), pods) {
+		t.Fatalf("get nodeinfo pods error,expected: %v,got: %v", pods, nodeinfo.Pods())
+	}
+}
+
+func TestPodsWithAffinity(t *testing.T) {
```
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Collapse the tests around PodWithAffinity and hasPodAffinityContraint.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go (hunk 3 above)
Comment (timothysc) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Or just initializing the resources where they are needed.

**Review by timothysc — COMMENTED** [review]
Some cleanup requests + probably need to take a look at the tests.

**Comment by timothysc** [comment]
@du2016 ping.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+		Spec: v1.NodeSpec{
+			ExternalID: "test",
+		},
+		Status: v1.NodeStatus{},
+	}
+}
+func TestNode(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	if !reflect.DeepEqual(nodeinfo.Node(), node) {
```
Comment (k82cn) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Check the pointer directly.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+	}
+}
+
+func TestPod(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	pod := makepod("foo", "", &v1.Affinity{})
+	pods := []*v1.Pod{pod}
+	nodeinfo.pods = pods
+	if !reflect.DeepEqual(nodeinfo.Pods(), pods) {
```
Comment (k82cn) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: What's case are we going to test here? Let's merge it with `AddPod`.

**Review by k82cn — COMMENTED** [review]
Suggest to remove the test on `get` methods, supposed we reserved 20% coverage for it

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+		t.Fatalf("nodeinfo check pod has no podaffinityconstraints error,expected: %v,got: %v", false, hasPodAffinityConstraints(podwithoutaffinity))
+	}
+}
+
+func TestaddPod(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	podwithaffinity := makepod("foo", "", &affnity)
+	nodeinfo.addPod(podwithaffinity)
+	if !reflect.DeepEqual(nodeinfo.podsWithAffinity, []*v1.Pod{podwithaffinity}) {
```
Comment (k82cn) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Add cases to check `RequestedResource` of node.

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+		t.Fatalf("nodeinfo add pod with affinity error,expected: %v,got: %v", []*v1.Pod{podwithaffinity}, nodeinfo.podsWithAffinity)
+	}
+	if !reflect.DeepEqual(nodeinfo.pods, []*v1.Pod{podwithaffinity}) {
+		t.Fatalf("nodeinfo add pod error,expected: %v,got: %v", []*v1.Pod{podwithaffinity}, nodeinfo.pods)
+	}
+
+}
+
+func TestremovePod(t *testing.T) {
```
Comment (k82cn) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: s/TestremovePod/TestRemovePod

File: plugin/pkg/scheduler/schedulercache/node_info_test.go [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	podwithaffinity := makepod("foo", "", &affnity)
+	podwithoutaffinity := makepod("foo", "", &v1.Affinity{})
+	nodeinfo.addPod(podwithaffinity)
+	nodeinfo.addPod(podwithoutaffinity)
```
Comment (k82cn) [inline plugin/pkg/scheduler/schedulercache/node_info_test.go]: Let's test exported method firstly; if key logic is not covered, then test the un-exported methods.

**Comment by k82cn** [comment]
@timothysc , it's Spring Festival in China now, so I think @du2016 is on vacation :).

**Comment by du2016** [comment]
@timothysc @k82cn thanks for reviewe,i will try to modify it.

**Comment by du2016** [comment]
i will open a new pr.

**Pull request closed without merging**